esp-metadata-generated/src/_generated_esp32.rs

```diff
@@ -1,4 +1,15 @@
 /// The name of the chip as `&str`
+///
+/// # Example
+///
+/// ```rust, no_run
+/// # {before_snippet}
+/// #
+/// use esp_hal::chip;
+/// let chip_name = chip!();
+/// #
+/// # {after_snippet}
```

Comment:
Why would we need any decoration around these macros, and what's the value of this example anyway?

Reply:
> This guideline should be applied within reason.

Reply:
My problem with our current approach of documenting these things is that we've read the title of the guideline, but not what it says. These examples should be informative, they should showcase why something can be useful. I think most people can deduce that if there is a macro, you can call that macro. Rustdoc even shows that it doesn't match on anything, and the documentation explains that it gives a `&str`. The guideline specifically calls out that just showing "how to call" something is a "poor example".